Prompt seems to expire/brake after one week

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

For a few months now, I've noticed that the Oh My Posh prompt expires or breaks after one week, by the 7th day. This seems to happen consistently, and across multiple opened terminals.

In the following images you can notice prompt from march 16 (that's the number '16', while the rest of the numbers are hour, minute and second), then a prompt from march 23, and finally a broken prompt from march 24:

<img width="620" height="64" alt="Image" src="https://github.com/user-attachments/assets/147e5c87-e89b-46f2-a325-09dee7f67c5d" />
<img width="620" height="72" alt="Image" src="https://github.com/user-attachments/assets/b9f74a6c-5034-4e58-a655-022714802304" />
<img width="620" height="56" alt="Image" src="https://github.com/user-attachments/assets/2b8e6826-fbc5-4dcb-8e76-4ea7b246cd0e" />

Is worth mentioning that at the time of those screenshots I had opened five (5) terminal tabs from the same day (march 16) and across all terminals the prompt broke by the 7th day (march 24).

The screenshots were taken when using ConEmu, but this issue has also happened on Windows Terminal.

Am I doing something wrong ?

### Theme

{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "template": "{{if .Env.PROMPT_PRE_LNBR}}\n{{end}}",
          "type": "text",
          "style": "plain"
        },
        {
          "template": " {{if .Env.VENV_SESSION}}{{.Env.VENV_SESSION}}{{else if .Env.WSYSTEM}}{{.Env.WSYSTEM}}{{else if .Env.MSYSTEM}}{{.Env.MSYSTEM}}{{end}}{{if .Root}}\uf132 {{end}}{{if not .Root}} {{end}}",
          "foreground": "#ffffff",
          "background": "#FF00FF",
          "type": "text",
          "style": "plain"
        },
        {
          "template": "\ue0b0",
          "foreground": "#FF00FF",
          "background": "#FFE50A",
          "type": "text",
          "style": "plain"
        },
        {
          "properties": {
            "home_icon": "\ue617",
            "mapped_locations": {
              "C:\\Program Files (x86)": "PROGRAMFILES32",
              "C:\\Program Files": "PROGRAMFILES",
              "C:\\Users": "\uf007",
              "C:\\Windows": "\ue70f",
              "D:\\mdnava": "\ue617",
              "D:\\mdnava\\Dropbox": "\uf16b",
              "D:\\mdnava\\Google Drive": "\ue731",
              "D:\\mdnava\\OneDrive": "\ue762",
              "D:\\mdnava\\OneDrive\\CodeX\\VS": "\ue70c VS",
              "D:\\mdnava\\OneDrive\\Command\\Git\\home\\mdnava": "\ue762\\Command\\Git\\home\\mdnava",
              "D:\\mdnava\\OneDrive\\Command\\Msys2\\home\\mdnava": "\ue762\\Command\\Msys2\\home\\mdnava",
              "D:\\mdnava\\OneDrive\\Images\\": "\uf1c5\\",
              "D:\\mdnava\\OneDrive\\Music\\": "\uf001\\"
            },
            "stack_count_enabled:": true,
            "style": "full"
          },
          "template": " {{ .Path }} {{ if matchP \"(?i)ON|True\" .Env.EXECTIME}}<#FFE50A,#ffffff>\ue0b0</>{{else}}<#FFE50A,#008000>\ue0b0</>{{end}}",
          "foreground": "#000000",
          "background": "#FFE50A",
          "type": "path"
        },
		{
		  "type": "executiontime",
		  "style": "powerline",
		  "foreground": "#c5c5c5",
		  "background": "#ffffff",
		  "template": "{{ if matchP \"(?i)ON|True\" .Env.EXECTIME}} {{ .FormattedMs }} <#ffffff,#008000>\ue0b0</>{{end}}",
		  "properties": {
			"threshold": 500,
			"style": "austin",
			"always_enabled": true
		  }
		},
		{
		  "properties": { "time_format": "15:04:05" },
		  "trailing_diamond": "\ue0b4",
		  "template": " <#c5c5c5>{{ .CurrentDate.Day }}.</>{{ .CurrentDate | date .Format }} ",
		  "foreground": "#FFFFFF",
		  "background": "#008000",
		  "type": "time",
		  "style": "diamond"
		},
        {
          "properties": {
            "text": " "
          },
          "type": "text",
          "style": "plain"
        }
      ]
    },
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "properties": {
            "branch_ahead_icon": "\u2191",
            "branch_behind_icon": "\u2193",
            "branch_gone_icon": "\u2262",
            "branch_icon": "\ue0a0",
            "branch_identical_icon": "\u2261",
            "fetch_stash_count": true,
            "fetch_status": true,
            "fetch_upstream_icon": true
          },
          "trailing_diamond": "\ue0bc",
          "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }} \uf0193 {{ .StashCount }}{{ end }} ",
          "foreground": "#193549",
          "background": "#95ffa4",
          "type": "git",
          "style": "diamond"
        },
        {
          "template": "❯ ",
          "foreground": "#ffffff",
          "type": "text"
        }
      ],
      "newline": true
    }
  ],
  "version": 3
}


### What OS are you seeing the problem on?

Windows

### Which shell are you using?

cmd

### Log output

```Shell
Version: 29.0.2

Shell: shell (clink v1.9.9.3fd236)

Prompt:

 SYSTEM   common                                                                      in shell at 09:11:32

Segments:

ConsoleTitle(true)                        -   0 ms
Session(true)                             -   0 ms
Path(true)                                -   1 ms
Git(false)                                -   1 ms
Root(false)                               -   0 ms
Status(true)                              -   1 ms
Node(false)                               -   9 ms
Go(false)                                 -   9 ms
Python(false)                             -   9 ms
Shell(true)                               -   0 ms
Time(true)                                -   0 ms

Run duration: 171.7745ms

Cache path: C:\Users\mdnava\AppData\Local\oh-my-posh

Config path: no --config set, using default built-in configuration

Logs:

[DEBUG] 09:11:32.444 debug.go:39 → logging enabled, raw mode: true
[DEBUG] 09:11:32.481 store.go:]:141 → (session) key not found: prompt_count_cache
[TRACE] 09:11:32.481 store.go:](session prompt_count_cache) - 17.1963ms
[TRACE] 09:11:32.482 terminal.go:setPromptCount() - 17.1963ms
[TRACE] 09:11:32.482 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.482 terminal.go:setPwd:102 → D:\mdnava\OneDrive\Projects\mnxNetTools\dns-tools\common
[TRACE] 09:11:32.482 terminal.go:setPwd() - 0s
[TRACE] 09:11:32.482 terminal.go:Init() - 18.1962ms
[DEBUG] 09:11:32.483 path_windows.go:PackageFamilyName:44 → failed to get PackageFamilyName
[TRACE] 09:11:32.483 path_windows.go:PackageFamilyName() - 1.0015ms
[TRACE] 09:11:32.483 path.go:Path() - 1.0015ms
[DEBUG] 09:11:32.483 store.go:init:94 → (device) loading INITVERSIONCMD
[DEBUG] 09:11:32.483 store.go:init:94 → (device) loading INITVERSIONPWSH
[DEBUG] 09:11:32.483 store.go:init:94 → (device) loading INITVERSIONBASH
[DEBUG] 09:11:32.483 store.go:init:94 → (device) loading DSC_SHELL
[DEBUG] 09:11:32.483 store.go:init:94 → (device) loading DSC_CONFIGURATION
[DEBUG] 09:11:32.483 store.go:init:94 → (device) loading INITVERSIONFISH
[TRACE] 09:11:32.483 store.go:init(device omp.cache) - 1.0015ms
[TRACE] 09:11:32.483 init.go:SessionID() - 0s
[TRACE] 09:11:32.483 path.go:Path() - 0s
[DEBUG] 09:11:32.497 store.go:init:94 → (session) loading prompt_count_cache
[DEBUG] 09:11:32.497 store.go:init:94 → (session) loading template_cache
[TRACE] 09:11:32.497 store.go:init(session cmd.99bd699c-2760-46eb-83ca-b94e2e1264d0.omp.cache) - 13.9967ms
[DEBUG] 09:11:32.498 store.go:]:141 → (device) key not found: reload
[TRACE] 09:11:32.498 store.go:](device reload) - 574.1µs
[DEBUG] 09:11:32.498 store.go:]:141 → (session) key not found: CONFIG
[TRACE] 09:11:32.498 store.go:](session CONFIG) - 591.5µs
[DEBUG] 09:11:32.498 gob.go:Get:39 → no cached config found
[DEBUG] 09:11:32.498 load.go:Parse:97 → no config file specified
[TRACE] 09:11:32.498 load.go:Parse() - 0s
[TRACE] 09:11:32.498 load.go:Load() - 0s
[TRACE] 09:11:32.498 gob.go:Get() - 591.5µs
[TRACE] 09:11:32.498 terminal.go:Shell() - 0s
[TRACE] 09:11:32.499 terminal.go:Flags() - 0s
[DEBUG] 09:11:32.499 store.go:]:154 → (session) found entry: template_cache - {map[Path:0xc000342000 Session:0xc000300690 Status:0xc000328050] map[] ~\OneDrive\Projects\mnxNetTools\dns-tools\common common  SYSTEM TESLA  cmd D:\mdnava\OneDrive\Projects\mnxNetTools\dns-tools\common windows 29.0.2 1 0 0 0 false false}
[TRACE] 09:11:32.499 store.go:](session template_cache) - 68.5µs
[TRACE] 09:11:32.499 cache.go:restoreCache() - 68.5µs
[DEBUG] 09:11:32.499 writer.go:Init:107 → terminal program: Unknown
[DEBUG] 09:11:32.499 writer.go:Init:108 → terminal shell: shell
[DEBUG] 09:11:32.499 terminal.go:Getenv:64 → NO DATA
[TRACE] 09:11:32.499 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[DEBUG] 09:11:32.499 store.go:]:141 → (device) key not found: accent_color
[TRACE] 09:11:32.499 store.go:](device accent_color) - 0s
[DEBUG] 09:11:32.499 terminal_windows.go:WindowsRegistryKeyValue:205 → ColorizationColor(DWORD): 0xC40078D7
[TRACE] 09:11:32.499 terminal_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 09:11:32.499 colors_windows.go:GetAccentColor() - 0s
[TRACE] 09:11:32.499 colors.go:SetAccentColor() - 0s
[TRACE] 09:11:32.499 terminal.go:Shell() - 0s
[DEBUG] 09:11:32.499 terminal.go:Getenv:64 → clink v1.9.9.3fd236
[TRACE] 09:11:32.499 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 09:11:32.499 debug.go:PrintDebug:24 → segment: Title
[DEBUG] 09:11:32.504 text.go:patchTemplate:149 → {{ .Shell }} in {{ .Folder }}
[TRACE] 09:11:32.504 terminal.go:Flags() - 0s
[DEBUG] 09:11:32.504 terminal.go:Getenv:64 → 16189
[TRACE] 09:11:32.504 terminal.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 09:11:32.504 terminal.go:Getenv:64 → NO DATA
[TRACE] 09:11:32.504 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 09:11:32.504 terminal.go:Flags() - 0s
[TRACE] 09:11:32.504 terminal.go:Flags() - 0s
[TRACE] 09:11:32.504 terminal.go:Flags() - 0s
[TRACE] 09:11:32.505 terminal.go:Flags() - 0s
[TRACE] 09:11:32.505 terminal.go:Flags() - 0s
[DEBUG] 09:11:32.505 home.go:12 → D:\mdnava
[TRACE] 09:11:32.505 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.505 home.go:12 → D:\mdnava
[DEBUG] 09:11:32.505 segment.go:Execute:172 → segment: Status
[TRACE] 09:11:32.505 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.505 home.go:12 → D:\mdnava
[DEBUG] 09:11:32.505 store.go:]:141 → (session) key not found: toggle_cache
[DEBUG] 09:11:32.505 segment.go:Execute:172 → segment: Path
[TRACE] 09:11:32.505 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.505 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.505 segment.go:Execute:172 → segment: Git
[DEBUG] 09:11:32.505 segment.go:isToggled:293 → no toggles found
[DEBUG] 09:11:32.505 store.go:]:141 → (session) key not found: toggle_cache
[TRACE] 09:11:32.505 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.505 segment.go:isToggled:293 → no toggles found
[TRACE] 09:11:32.505 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.505 segment.go:isToggled:293 → no toggles found
[TRACE] 09:11:32.505 terminal.go:Flags() - 0s
[DEBUG] 09:11:32.505 home.go:12 → D:\mdnava
[TRACE] 09:11:32.505 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.505 segment.go:Execute:172 → segment: Session
[DEBUG] 09:11:32.505 store.go:]:141 → (session) key not found: toggle_cache
[TRACE] 09:11:32.505 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.505 segment.go:isToggled:293 → no toggles found
[TRACE] 09:11:32.505 terminal.go:StatusCodes() - 0s
[DEBUG] 09:11:32.505 terminal.go:Getenv:64 → NO DATA
[TRACE] 09:11:32.505 terminal.go:Getenv(SSH_CONNECTION) - 0s
[DEBUG] 09:11:32.505 terminal.go:Getenv:64 → NO DATA
[TRACE] 09:11:32.505 terminal.go:Getenv(SSH_CLIENT) - 0s
[DEBUG] 09:11:32.505 map.go:Bool:133 → disable_with_jj: false
[DEBUG] 09:11:32.505 map.go:Bool:133 → display_cygpath: false
[TRACE] 09:11:32.505 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.505 map.go:String:79 → status_template: {{ .Code }}
[TRACE] 09:11:32.505 terminal.go:Flags() - 0s
[DEBUG] 09:11:32.505 home.go:12 → D:\mdnava
[TRACE] 09:11:32.505 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.505 segment.go:Execute:172 → segment: Root
[DEBUG] 09:11:32.505 store.go:]:141 → (session) key not found: toggle_cache
[TRACE] 09:11:32.505 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.505 segment.go:isToggled:293 → no toggles found
[DEBUG] 09:11:32.505 text.go:patchTemplate:149 → {{ .Code }}
[DEBUG] 09:11:32.505 map.go:Bool:141 → always_enabled: true
[DEBUG] 09:11:32.506 text.go:patchTemplate:149 →  {{ if .Data.SSHSession }} {{ end }}{{ .UserName }}
[TRACE] 09:11:32.506 separator.go:Separator() - 0s
[TRACE] 09:11:32.506 terminal.go:Shell() - 0s
[TRACE] 09:11:32.506 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.506 terminal_windows.go:Root() - 0s
[DEBUG] 09:11:32.506 map.go:Bool:133 → mapped_locations_enabled: true
[DEBUG] 09:11:32.506 map.go:String:79 → windows_registry_icon: 
[DEBUG] 09:11:32.506 map.go:String:79 → windows_registry_icon: 
[DEBUG] 09:11:32.506 home.go:12 → D:\mdnava
[TRACE] 09:11:32.506 separator.go:Separator() - 0s
[TRACE] 09:11:32.506 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.506 map.go:String:79 → home_icon: ~
[DEBUG] 09:11:32.506 map.go:KeyValueMap:208 → mapped_locations: map[]
[TRACE] 09:11:32.506 separator.go:Separator() - 0s
[TRACE] 09:11:32.506 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.506 separator.go:Separator() - 0s
[DEBUG] 09:11:32.506 terminal.go:ResolveSymlink:195 → D:\mdnava\OneDrive\Projects\mnxNetTools\dns-tools\common
[TRACE] 09:11:32.506 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.506 terminal.go:ResolveSymlink(D:\mdnava\OneDrive\Projects\mnxNetTools\dns-tools\common) - 622.2µs
[TRACE] 09:11:32.506 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.506 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.506 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.506 map.go:String:79 → gitdir_format:
[DEBUG] 09:11:32.506 map.go:String:83 → style: folder
[DEBUG] 09:11:32.506 map.go:StringArray:220 → cycle: []
[DEBUG] 09:11:32.506 map.go:String:79 → folder_separator_template:
[DEBUG] 09:11:32.506 map.go:String:79 → folder_separator_icon: \
[DEBUG] 09:11:32.506 map.go:Bool:133 → cycle_folder_separator: false
[DEBUG] 09:11:32.506 map.go:String:79 → folder_format: %s
[DEBUG] 09:11:32.506 map.go:String:79 → edge_format: %s
[DEBUG] 09:11:32.506 map.go:String:79 → left_format: %s
[DEBUG] 09:11:32.506 map.go:String:79 → right_format: %s
[TRACE] 09:11:32.506 terminal.go:Flags() - 0s
[TRACE] 09:11:32.506 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.506 terminal.go:StackCount() - 0s
[ERROR] 09:11:32.506 terminal.go:HasParentFilePath:464 → GetFileAttributesEx .git: The system cannot find the file specified.
[TRACE] 09:11:32.506 terminal.go:HasParentFilePath(.git) - 1.1504ms
[DEBUG] 09:11:32.507 win32_windows.go:isWriteable:263 → current user is member of S-1-5-32-544
[DEBUG] 09:11:32.507 win32_windows.go:isWriteable:271 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 09:11:32.507 win32_windows.go:isWriteable:273 → user has write access
[TRACE] 09:11:32.507 terminal_windows.go:DirIsWritable() - 534.1µs
[DEBUG] 09:11:32.507 text.go:patchTemplate:149 →   {{ path .Data.Path .Data.Location }}
[DEBUG] 09:11:32.509 text.go:patchTemplate:149 →  {{ if gt .Code 0 }}{{ else }}{{ end }}
[DEBUG] 09:11:32.509 text.go:patchTemplate:149 → {{ if gt .Code 0 }}p:red{{ end }}
[DEBUG] 09:11:32.509 text.go:patchTemplate:149 → {{ if gt .Code 0 }}p:red{{ end }}
[TRACE] 09:11:32.509 terminal.go:Shell() - 0s
[TRACE] 09:11:32.509 terminal.go:Flags() - 0s
[TRACE] 09:11:32.509 terminal.go:Flags() - 0s
[TRACE] 09:11:32.509 terminal.go:Flags() - 0s
[DEBUG] 09:11:32.509 home.go:12 → D:\mdnava
[TRACE] 09:11:32.509 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.509 segment.go:Execute:172 → segment: Time
[TRACE] 09:11:32.509 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.509 store.go:]:141 → (session) key not found: toggle_cache
[DEBUG] 09:11:32.509 segment.go:Execute:172 → segment: Node
[TRACE] 09:11:32.509 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.509 store.go:]:141 → (session) key not found: toggle_cache
[DEBUG] 09:11:32.509 home.go:12 → D:\mdnava
[DEBUG] 09:11:32.509 segment.go:isToggled:293 → no toggles found
[TRACE] 09:11:32.509 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.509 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.509 segment.go:isToggled:293 → no toggles found
[DEBUG] 09:11:32.509 segment.go:Execute:172 → segment: Python
[DEBUG] 09:11:32.509 map.go:String:79 → time_format: 15:04:05
[DEBUG] 09:11:32.509 store.go:]:141 → (session) key not found: toggle_cache
[TRACE] 09:11:32.509 terminal.go:Flags() - 0s
[TRACE] 09:11:32.509 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.509 segment.go:isToggled:293 → no toggles found
[TRACE] 09:11:32.509 terminal.go:Flags() - 0s
[DEBUG] 09:11:32.509 home.go:12 → D:\mdnava
[TRACE] 09:11:32.509 terminal.go:GOOS() - 0s
[DEBUG] 09:11:32.509 segment.go:Execute:172 → segment: Shell
[DEBUG] 09:11:32.509 store.go:]:141 → (session) key not found: toggle_cache
[DEBUG] 09:11:32.509 map.go:String:83 → display_mode: files
[TRACE] 09:11:32.509 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.509 segment.go:isToggled:293 → no toggles found
[DEBUG] 09:11:32.509 map.go:KeyValueMap:208 → mapped_shell_names: map[]
[DEBUG] 09:11:32.509 home.go:12 → D:\mdnava
[TRACE] 09:11:32.509 terminal.go:Shell() - 0s
[TRACE] 09:11:32.509 terminal.go:GOOS() - 0s
[TRACE] 09:11:32.509 terminal.go:Flags() - 0s
[DEBUG] 09:11:32.509 segment.go:Execute:172 → segment: Go
[DEBUG] 09:11:32.509 store.go:]:141 → (session) key not found: toggle_cache
[TRACE] 09:11:32.509 store.go:](session toggle_cache) - 0s
[DEBUG] 09:11:32.509 segment.go:isToggled:293 → no toggles found
[DEBUG] 09:11:32.517 map.go:StringArray:220 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 09:11:32.517 map.go:StringArray:220 → folders: [.venv venv virtualenv venv-win pyenv-win]
[DEBUG] 09:11:32.517 map.go:Bool:133 → home_enabled: false
[DEBUG] 09:11:32.517 map.go:StringArray:220 → extensions: [*.go go.mod go.sum go.work go.work.sum]
[DEBUG] 09:11:32.517 map.go:StringArray:220 → folders: []
[DEBUG] 09:11:32.517 map.go:Bool:141 → home_enabled: false
[DEBUG] 09:11:32.517 map.go:StringArray:220 → folders: []
[DEBUG] 09:11:32.517 map.go:Bool:133 → home_enabled: false
[DEBUG] 09:11:32.517 home.go:12 → D:\mdnava
[DEBUG] 09:11:32.518 home.go:12 → D:\mdnava
[DEBUG] 09:11:32.518 map.go:String:79 → display_mode: files
[DEBUG] 09:11:32.518 home.go:12 → D:\mdnava
[DEBUG] 09:11:32.518 map.go:String:83 → display_mode: files
[DEBUG] 09:11:32.518 map.go:Bool:141 → fetch_virtual_env: false
[DEBUG] 09:11:32.518 map.go:Bool:141 → fetch_package_manager: false
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(*.go) - 532.7µs
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(*.py) - 532.7µs
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(go.mod) - 0s
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(*.js) - 0s
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(*.ipynb) - 0s
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(go.sum) - 0s
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(*.ts) - 0s
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(pyproject.toml) - 0s
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(go.work) - 0s
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(venv.bak) - 0s
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(go.work.sum) - 0s
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[DEBUG] 09:11:32.518 map.go:StringArray:220 → tooling: [mod go]
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(.nvmrc) - 0s
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(pnpm-workspace.yaml) - 0s
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(.pnpmfile.cjs) - 0s
[DEBUG] 09:11:32.518 terminal.go:HasFilesInDir:151 → false
[TRACE] 09:11:32.518 terminal.go:HasFilesInDir(.vue) - 0s
[DEBUG] 09:11:32.518 map.go:StringArray:220 → tooling: [node]
[DEBUG] 09:11:32.518 terminal.go:HasFolder:180 → false
[TRACE] 09:11:32.518 terminal.go:HasFolder(.venv) - 0s
[DEBUG] 09:11:32.519 terminal.go:HasFolder:180 → false
[TRACE] 09:11:32.519 terminal.go:HasFolder(venv) - 505.4µs
[DEBUG] 09:11:32.519 terminal.go:HasFolder:180 → false
[TRACE] 09:11:32.519 terminal.go:HasFolder(virtualenv) - 36.6µs
[DEBUG] 09:11:32.519 terminal.go:HasFolder:180 → false
[TRACE] 09:11:32.519 terminal.go:HasFolder(venv-win) - 0s
[DEBUG] 09:11:32.519 terminal.go:HasFolder:180 → false
[TRACE] 09:11:32.519 terminal.go:HasFolder(pyenv-win) - 0s
[DEBUG] 09:11:32.519 map.go:StringArray:220 → tooling: [pyenv python python3 py]
[DEBUG] 09:11:32.519 text.go:patchTemplate:149 → in <p:blue><b>{{ .Data.Name }}</b></>
[DEBUG] 09:11:32.519 text.go:patchTemplate:149 → at <p:blue><b>{{ .Data.CurrentDate | date "15:04:05" }}</b></>
[TRACE] 09:11:32.519 terminal.go:Shell() - 0s
[TRACE] 09:11:32.519 terminal.go:Flags() - 0s
[TRACE] 09:11:32.519 terminal.go:Shell() - 0s
[DEBUG] 09:11:32.519 terminal_windows.go:TerminalWidth:99 → terminal width: 114
[TRACE] 09:11:32.519 terminal_windows.go:TerminalWidth() - 59.5µs
[TRACE] 09:11:32.519 path.go:Path() - 0s
```

--- a/.github/agents/architecture.md
+++ b/.github/agents/architecture.md
@@ -1,0 +1,155 @@
+---
+name: Architecture and Design
+description: >-
+  Cross-language architectural guidance for designing scalable, maintainable,
+  and performant code. Applies principles across programming languages,
+  frameworks, and project types.
+---
+
+## Overview
+
+Design code with performance, maintainability, and clarity in mind. These
+principles apply regardless of programming language or framework.
+
+## Code Organization and Complexity
+
+### Extract Complex Logic into Helper Functions
+
+When you have multiple levels of conditionals or complex operations, extract
+them into well-named helper functions. This reduces nesting and clarifies
+intent.
+
+**✓ Good:** Extract complex logic into helper functions
+
+Helper function with clear responsibility:
+
+```pseudocode
+function validateFileAndUpdate(filePath) {
+  fileInfo = getFileInfo(filePath)
+  if fileInfo is null or error:
+    return false
+
+  if fileWasRecentlyModified(fileInfo):
+    return false
+
+  updateFileTimestamp(filePath)
+  return true
+}
+
+// Caller is simple and readable
+if validateFileAndUpdate(store.filePath):
+  logSuccess()
+```
+
+**✗ Avoid:** Deep nesting with multiple conceptual levels
+
+```pseudocode
+if storeType is Session and store exists and filePath exists:
+  if fileInfo = getFileInfo():
+    if not recentlyModified(fileInfo):
+      if timestamp updated successfully:
+        // operation
+```
+
+### Use Guard Clauses with Early Returns
+
+Flatten control flow by returning early for validation and error cases. This
+moves the happy path to the left and reduces nesting.
+
+**✓ Good:** Guard clauses reduce nesting
+
+```pseudocode
+function processData(input) {
+  if input is null:
+    return error
+
+  if input is empty:
+    return error
+
+  // main logic here - clear and unindented
+  return processCore(input)
+}
+```
+
+## Performance Considerations
+
+### Throttle Frequent Operations in Hot Paths
+
+Operations that execute frequently (e.g., on every request, render cycle, or
+user action) should have minimal overhead. Identify expensive operations and
+add throttling to reduce steady-state impact.
+
+**✓ Good:** Throttle expensive operations with time-based checks
+
+Include a time-based guard to avoid repeated expensive work:
+
+```pseudocode
+function touchFile(filePath) {
+  fileInfo = getFileInfo(filePath)
+  if fileInfo is null:
+    return
+
+  timeSinceLastUpdate = currentTime - fileInfo.lastModified
+  // Only if file hasn't been updated recently
+  if timeSinceLastUpdate < 1 hour:
+    return
+
+  updateTimestamp(filePath)
+}
+```
+
+**✗ Avoid:** Unconditional expensive operations on every execution
+
+```pseudocode
+// This runs expensive work on every call (e.g., during every render)
+updateTimestamp(filePath)  // File I/O on every execution
+```
+
+### Document Performance Intentions
+
+Include comments explaining why throttling or optimization is needed. This
+helps reviewers understand the performance tradeoffs.
+
+```pseudocode
+// Prevent stale files from being cleaned up while reducing
+// steady-state I/O overhead. Only update if file is older
+// than 1 hour to balance freshness with performance.
+if timeSinceUpdate > 1 hour:
+  updateTimestamp(filePath)
+```
+
+## Error Handling
+
+- Check for errors and validate inputs early, before expensive operations
+- Return or fail fast to avoid deeply nested success paths
+- Each error should include sufficient context for debugging
+- Early returns make the happy path obvious and easier to follow
+
+## Code Review Checklist
+
+When reviewing code:
+
+- **Nesting depth:** Flag functions with 3+ levels of indentation as
+  refactoring candidates
+- **Hot path operations:** Verify frequent operations minimize I/O,
+  allocations, and expensive calls
+- **Early returns:** Confirm guard clauses validate inputs before main logic
+- **Comments:** Check that performance-critical code explains the tradeoff,
+  not just the mechanics
+- **Extraction opportunities:** Identify deeply nested conditions that could
+  become helpers
+
+## Core Principles
+
+1. **Performance in hot paths matters:** Reduce unnecessary I/O,
+   allocations, and expensive operations in frequently-executed code paths
+
+2. **Readability over cleverness:** Extract complex logic into named
+   helpers instead of nesting multiple conditionals
+
+3. **Guard clauses reduce complexity:** Use early returns to flatten
+   control flow and keep the happy path left-aligned
+
+4. **Comments explain why, not what:** Document performance tradeoffs,
+   business logic, and non-obvious decisions—let code structure explain
+   the mechanics

--- a/.github/workflows/code.yml
+++ b/.github/workflows/code.yml
@@ -40,4 +40,4 @@ jobs:
         go install golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest
         modernize "./..."
     - name: Unit Tests
-      run: go test -v ./...
+      run: go test "./..."

--- a/.markdownlint-cli2.yaml
+++ b/.markdownlint-cli2.yaml
@@ -8,4 +8,5 @@ gitignore: true
 ignores:
   - node_modules/
   - .github/agents/segment-docs.md
+  - .github/agents/architecture.md
   - .github/PULL_REQUEST_TEMPLATE.md

--- a/AGENTS.md
+++ b/AGENTS.md
@@ -17,6 +17,16 @@ When editing Go files (`*.go`):
   - Run `gofmt` (or `go fmt`) and organize imports.
   - Run `golangci-lint run` at the repository root and address findings.
 
+## Architecture and Design
+
+When working on code that impacts performance, maintainability, or system
+design:
+
+- Consult the **Architecture and Design** agent for cross-language guidance on
+  code organization, complexity management, and performance considerations.
+- Key principles: extract complex logic into helpers, use guard clauses and
+  early returns, throttle frequent operations in hot paths.
+
 ## Markdown
 
 When editing Markdown (`*.md`, `*.mdx`):

--- a/src/cache/store.go
+++ b/src/cache/store.go
@@ -3,6 +3,7 @@ package cache
 import (
 	"encoding/gob"
 	"fmt"
+	"os"
 	"path/filepath"
 	"strings"
 	"time"
@@ -96,11 +97,32 @@ func (s Store) init(filePath string, persist bool) {
 	}
 }
 
+// touchSessionFile updates the session file's modification time if it's older than 1 hour.
+// This prevents stale session cache files from being cleaned up while reducing steady-state overhead.
+func touchSessionFile(filePath string) {
+	info, err := os.Stat(filePath)
+	if err != nil {
+		return
+	}
+
+	if time.Since(info.ModTime()) <= time.Hour {
+		return
+	}
+
+	if err := os.Chtimes(filePath, time.Now(), time.Now()); err != nil {
+		log.Error(err)
+	}
+}
+
 func (s Store) close() {
 	defer log.Trace(time.Now(), string(s))
 
 	store := s.get()
 	if store == nil || !store.persist || !store.dirty {
+		if s == Session && store != nil && store.filePath != "" {
+			touchSessionFile(store.filePath)
+		}
+
 		log.Debugf("(%s) not persisting", string(s))
 		return
 	}

--- a/src/color/colors.go
+++ b/src/color/colors.go
@@ -49,7 +49,7 @@ func (c *Set) String() string {
 }
 
 func (c *Set) ParseString(colors string) {
-	parts := strings.Split(colors, "|")
+	parts := strings.SplitN(colors, "|", 3)
 	if len(parts) != 2 {
 		return
 	}

--- a/src/segments/aws.go
+++ b/src/segments/aws.go
@@ -80,7 +80,7 @@ func (a *Aws) getConfigFileInfo() {
 		}
 
 		if sectionActive && strings.HasPrefix(line, "region") {
-			splitted := strings.Split(line, "=")
+			splitted := strings.SplitN(line, "=", 3)
 			if len(splitted) >= 2 {
 				a.Region = strings.TrimSpace(splitted[1])
 				break

--- a/src/segments/git.go
+++ b/src/segments/git.go
@@ -785,7 +785,7 @@ func (g *Git) setStatus() {
 
 		if strings.HasPrefix(line, BRANCHSTATUS) && len(line) > len(BRANCHSTATUS) {
 			status := line[len(BRANCHSTATUS):]
-			splitted := strings.Split(status, " ")
+			splitted := strings.SplitN(status, " ", 3)
 			if len(splitted) >= 2 {
 				g.Ahead, _ = strconv.Atoi(splitted[0])
 				behind, _ := strconv.Atoi(splitted[1])

--- a/src/segments/git.go
+++ b/src/segments/git.go
@@ -312,8 +312,7 @@ func (g *Git) StashCount() int {
 		return 0
 	}
 
-	lines := strings.Split(stashContent, "\n")
-	g.stashCount = len(lines)
+	g.stashCount = strings.Count(stashContent, "\n") + 1 // +1: fileContent() trims
 	return g.stashCount
 }
 

--- a/src/segments/jujutsu.go
+++ b/src/segments/jujutsu.go
@@ -79,14 +79,14 @@ func (jj *Jujutsu) ClosestBookmarks() string {
 		return ""
 	}
 
-	lines := strings.Split(statusString, "\n")
+	line, _, _ := strings.Cut(statusString, "\n")
 
-	if !jj.options.Bool(FetchAhead, false) || len(lines[0]) == 0 {
-		return lines[0]
+	if !jj.options.Bool(FetchAhead, false) || len(line) == 0 {
+		return line
 	}
 
 	aheadIcon := jj.options.String(AheadIcon, "\u21e1")
-	marks := strings.Split(lines[0], " ")
+	marks := strings.Split(line, " ")
 	// String to return for status
 	var endString strings.Builder
 
@@ -97,7 +97,7 @@ func (jj *Jujutsu) ClosestBookmarks() string {
 
 	aheadString, err := jj.getJujutsuCommandOutput("log", "--no-graph", "-T", "'.'", "-r", rangeString)
 	if err != nil {
-		return lines[0]
+		return line
 	}
 
 	aheadCounter := len(aheadString)

--- a/src/segments/mercurial.go
+++ b/src/segments/mercurial.go
@@ -107,7 +107,7 @@ func (hg *Mercurial) setMercurialStatus() {
 		return
 	}
 
-	idSplit := strings.Split(idString, "|")
+	idSplit := strings.SplitN(idString, "|", 6)
 	if len(idSplit) != 5 {
 		return
 	}

--- a/src/segments/python.go
+++ b/src/segments/python.go
@@ -177,7 +177,7 @@ func (p *Python) pyenvVersion() (string, error) {
 	}
 
 	// override virtualenv if pyenv set one
-	parts := strings.Split(shortPath, string(filepath.Separator))
+	parts := strings.SplitN(shortPath, string(filepath.Separator), 4)
 	if len(parts) > 2 && p.canUseVenvName(parts[2]) {
 		p.Venv = parts[2]
 	}

--- a/src/segments/winget.go
+++ b/src/segments/winget.go
@@ -2,6 +2,7 @@ package segments
 
 import (
 	"strings"
+	"unicode"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/runtime"
 )
@@ -48,10 +49,9 @@ func (w *WinGet) parseWinGetOutput(output string) []WinGetPackage {
 	var packages []WinGetPackage
 
 	lines := strings.Split(output, "\n")
-	var headerLine string
 	separatorIndex := -1
 
-	// First pass: find header line and separator
+	// Find the separator line
 	for i, line := range lines {
 		if line == "" {
 			continue
@@ -61,11 +61,6 @@ func (w *WinGet) parseWinGetOutput(output string) []WinGetPackage {
 			separatorIndex = i
 			break
 		}
-
-		// Store the header line to determine column positions
-		if strings.Contains(line, "Id") && strings.Contains(line, "Version") {
-			headerLine = line
-		}
 	}
 
 	// If no separator found, return empty
@@ -73,15 +68,18 @@ func (w *WinGet) parseWinGetOutput(output string) []WinGetPackage {
 		return packages
 	}
 
-	// Determine column positions from header (calculated once)
-	nameIndex := strings.Index(headerLine, "Name")
-	idIndex := strings.Index(headerLine, "Id") - nameIndex
-	versionIndex := strings.Index(headerLine, "Version") - nameIndex
-	availableIndex := strings.Index(headerLine, "Available") - nameIndex
-	sourceIndex := strings.Index(headerLine, "Source") - nameIndex
+	// The header line is right before the separator
+	if separatorIndex == 0 {
+		return packages
+	}
 
-	// If we can't find column positions, return empty
-	if idIndex < 0 || versionIndex < 0 || availableIndex < 0 {
+	headerLine := lines[separatorIndex-1]
+
+	// Find column positions by detecting transitions from non-letter to letter
+	columnIndices := findColumnIndices(headerLine)
+
+	// We need at least 4 columns: Name, Id, Version, Available
+	if len(columnIndices) < 4 {
 		return packages
 	}
 
@@ -98,30 +96,16 @@ func (w *WinGet) parseWinGetOutput(output string) []WinGetPackage {
 			continue
 		}
 
-		// Skip lines that are too short
-		if len(line) < availableIndex {
+		// Skip lines that are too short to contain all fields
+		if len(line) < columnIndices[3] {
 			continue
 		}
 
 		// Extract fields using column positions
-		name := strings.TrimSpace(line[:idIndex])
-
-		var id, current, available string
-		if len(line) >= versionIndex {
-			id = strings.TrimSpace(line[idIndex:versionIndex])
-		}
-		if len(line) >= availableIndex {
-			current = strings.TrimSpace(line[versionIndex:availableIndex])
-		}
-
-		if sourceIndex > 0 && len(line) >= sourceIndex {
-			available = strings.TrimSpace(line[availableIndex:sourceIndex])
-		}
-
-		if available == "" && len(line) > availableIndex {
-			// No source column or line is shorter
-			available = strings.TrimSpace(line[availableIndex:])
-		}
+		name := extractField(line, columnIndices[0], columnIndices[1])
+		id := extractField(line, columnIndices[1], columnIndices[2])
+		current := extractField(line, columnIndices[2], columnIndices[3])
+		available := extractField(line, columnIndices[3], columnIndices[4])
 
 		// Skip if essential fields are empty
 		if name == "" || id == "" {
@@ -139,4 +123,43 @@ func (w *WinGet) parseWinGetOutput(output string) []WinGetPackage {
 	}
 
 	return packages
+}
+
+func findColumnIndices(headerLine string) []int {
+	var allIndices []int
+	prevWasLetter := false
+
+	for i, char := range headerLine {
+		isLetter := unicode.IsLetter(char)
+
+		if isLetter && !prevWasLetter {
+			// Found transition from non-letter to letter - this is a column start
+			allIndices = append(allIndices, i)
+		}
+
+		prevWasLetter = isLetter
+	}
+
+	// Get the last 5 column positions (Name, Id, Version, Available, Source)
+	// This handles cases where there's garbage before the actual header
+	lastFive := allIndices[len(allIndices)-5:]
+
+	// Make all positions relative to the first column position
+	firstColPos := lastFive[0]
+	relativeIndices := make([]int, len(lastFive))
+	for i, pos := range lastFive {
+		relativeIndices[i] = pos - firstColPos
+	}
+
+	return relativeIndices
+}
+
+func extractField(line string, startIndex, endIndex int) string {
+	if startIndex >= len(line) {
+		return ""
+	}
+	if endIndex > len(line) {
+		endIndex = len(line)
+	}
+	return strings.TrimSpace(line[startIndex:endIndex])
 }

--- a/src/segments/winget_test.go
+++ b/src/segments/winget_test.go
@@ -165,6 +165,20 @@ Python 3.11        Python.Python.3.11          3.11.0    3.11.5    winget
 				Available: "3.11.5",
 			},
 		},
+		{
+			Case: "Spanish headers (localization test)",
+			Output: `Nombre            Id                          Vers      Dispo  Origen
+-----------------------------------------------------------------------------------
+Python 3.11        Python.Python.3.11          3.11.0    3.11.5    winget
+Node.js            OpenJS.NodeJS               18.0.0    18.12.1   winget`,
+			ExpectedCount: 2,
+			ExpectedFirst: WinGetPackage{
+				Name:      "Python 3.11",
+				ID:        "Python.Python.3.11",
+				Current:   "3.11.0",
+				Available: "3.11.5",
+			},
+		},
 	}
 
 	for _, tc := range cases {

--- a/themes/multiverse-neon.omp.json
+++ b/themes/multiverse-neon.omp.json
@@ -57,9 +57,10 @@
             "display_mode": "files",
             "fetch_package_manager": true,
             "fetch_version": true,
+            "bun_icon": "<#e5dbcc>\ue76f bun</>",
             "npm_icon": "<#e5272d>\ue616 npm</> ",
-            "yarn_icon": "<#37aee5>\ue6a7 yarn</> ",
-            "pnpm_icon": "<#e5a100>\udb80\udec1 pnpm</> "
+            "pnpm_icon": "<#e5a100>\ue865 pnpm</> ",
+            "yarn_icon": "<#37aee5>\ue6a7 yarn</> "
           },
           "template": "{{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{ end }}\ue718 {{ .Full }}"
         },

